src/Modules/ContentBlocks/Domain/ContentBlock/ContentBlockRepository.php

```diff
@@ -0,0 +1,179 @@
+    }
+
+    public function removeByRevisionId(int $revisionId): void
+    {
+        $contentBlock = $this->findOneby(['revisionId' => $revisionId]);
+
+        if ($contentBlock === null) {
+            return;
+        }
+
+        // get all versions
+        $versions = $this->findBy(['id' => $contentBlock->getId(), 'locale' => $contentBlock->getLocale()]);
+
+        $entityManager = $this->getEntityManager();
+        foreach ($versions as $version) {
+            $entityManager->remove($version);
+        }
+
+        $entityManager->flush();
+    }
+
+    public function getNextIdForLocale(Locale $locale): int
+    {
+        return (int) $this->getEntityManager()
+                ->createQueryBuilder()
+                ->select('MAX(cb.id) as id')
+                ->from(ContentBlock::class, 'cb')
+                ->where('cb.locale = :locale')
+                ->setParameter('locale', $locale)
+                ->getQuery()
+                ->getSingleScalarResult() + 1;
+    }
+
+    public function findForIdAndLocale(int $id, string $language): ?ContentBlock
+    {
+        return $this->createQueryBuilder('cb')
+            ->andWhere('cb.id = :id')
+            ->andWhere('cb.status = :active')
+            ->andWhere('cb.locale = :locale')
+            ->andWhere('cb.isHidden = :false')
+            ->setParameter('id', $id)
+            ->setParameter('active', Status::Active)
+            ->setParameter('locale', Locale::from($language))
+            ->setParameter('false', false)
+            ->getQuery()
+            ->getOneOrNullResult();
+    }
+
+    private function updateWidget(ContentBlock $contentBlock): void
+    {
+        /** @var Block $block */
+        $block = $this->blockRepository->findOneBy(['id' => $contentBlock->getExtraId()]);
+
+        $block->getSettings()->add([
+            'label' => $contentBlock->getTitle(),
+            'content_block_id' => $contentBlock->getId()
+        ]);
+
+        if ($contentBlock->isHidden()) {
+            $block->hide();
+        } else {
+            $block->show();
+        }
+
+        $this->blockRepository->save($block);
+    }
+
+    /**
+     * @return array<ContentBlock>
+     */
+    public function getRevisionsForContentBlock(ContentBlock $contentBlock): array
+    {
+        return $this->createQueryBuilder('cb')
+            ->andWhere('cb.id = :id')
+            ->andWhere('cb.locale = :locale')
+            ->andWhere('cb.status = :archived')
+            ->addOrderBy('cb.updatedOn', 'ASC')
+            ->setParameter('id', $contentBlock->getId())
+            ->setParameter('locale', $contentBlock->getLocale())
+            ->setParameter('archived', Status::Archived)
+            ->getQuery()
+            ->getResult();
+    }
+
+    public function isContentBlockInUse(ContentBlock $contentBlock): bool
+    {
+        $result = $this->getEntityManager()->createQueryBuilder()
```

Comment:
Shouldn't this be casted to `(int)`?

Reply:
From what I see it should already be an it when it is returned so no cast needed